From: =?utf-8?q?Ant=C3=B3nio_Fernandes?= <antoniof@gnome.org>
Date: Wed, 30 Dec 2020 17:41:01 +0000
Subject: mime-actions: Drop dead code path

Now that the async files opening path is only used while sandboxed,
the conditions for these if blocks (not being sanboxed) is never met.

Origin: upstream, 40.rc, commit:17285c25834cea9fd7b3fb7b155127db116004af
---
 src/nautilus-mime-actions.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/nautilus-mime-actions.c b/src/nautilus-mime-actions.c
index 46188ad..c499111 100644
--- a/src/nautilus-mime-actions.c
+++ b/src/nautilus-mime-actions.c
@@ -90,7 +90,6 @@ typedef struct
 {
     ActivateParameters *activation_params;
     GQueue *uris;
-    GQueue *unhandled_uris;
 } ApplicationLaunchAsyncParameters;
 
 /* Microsoft mime types at https://blogs.msdn.microsoft.com/vsofficedeveloper/2008/05/08/office-2007-file-format-mime-types-for-http-content-streaming-2/ */
@@ -982,7 +981,6 @@ activation_parameters_free (ActivateParameters *parameters)
 static void
 application_launch_async_parameters_free (ApplicationLaunchAsyncParameters *parameters)
 {
-    g_queue_free (parameters->unhandled_uris);
     g_queue_free (parameters->uris);
     activation_parameters_free (parameters->activation_params);
 
@@ -1520,10 +1518,6 @@ launch_default_for_uris_callback (GObject      *source_object,
     uri = g_queue_pop_head (params->uris);
 
     nautilus_launch_default_for_uri_finish (res, &error);
-    if (!is_sandboxed () && error != NULL && error->code != G_IO_ERROR_CANCELLED)
-    {
-        g_queue_push_tail (params->unhandled_uris, uri);
-    }
 
     if (!g_queue_is_empty (params->uris))
     {
@@ -1535,11 +1529,6 @@ launch_default_for_uris_callback (GObject      *source_object,
     }
     else
     {
-        while ((uri = g_queue_pop_head (params->unhandled_uris)) != NULL)
-        {
-            application_unhandled_uri (activation_params, uri);
-        }
-
         application_launch_async_parameters_free (params);
     }
 }
